File: Visitor/python-source/rooms/LivingRoom.py
from __future__ import annotations

from typing import TYPE_CHECKING

from .ElementRoom import ElementRoom

if TYPE_CHECKING:
    from workers import Visitor


class LivingRoom(ElementRoom):
    def accept(self, visitor: Visitor) -> None:
        visitor.visit_living_room(self)

    def get_name(self) -> str:
        return "LivingRoom"

File: JavaSource/src/main/java/ru/akhcheck/patterns/visitor/rooms/LivingRoom.java
package ru.akhcheck.patterns.visitor.rooms;

import ru.akhcheck.patterns.visitor.workers.Visitor;

/** Гостиная: здесь не бывает плитки и воды. */
public class LivingRoom extends ElementRoom {
    @Override
    public void accept(Visitor visitor) {
        visitor.visit(this);
    }

    @Override
    public String getName() {
        return "LivingRoom";
    }
}
